test/unit/lodash.spec.js

```diff
@@ -0,0 +1,522 @@
+                });
+            });
+        });
+
+
+        describe('mean', function () {
+
+            var params;
+
+            beforeEach(function () {
+                params = datasets.mean();
+            });
+
+            it('should find mean value (3.25) in user array', function () {
+                expect(params).to.have.length(1);
+                _(params[0]).forEach(function (value) {
+                    expect(typeof value).to.eql('number');
+                });
+                expect(_.mean.apply(_, params)).to.eql(3.25);
+            });
+        });
+
+        describe('meanBy', function () {
+            describe('meanBy - Array', function () {
+
+                var params;
+
+                beforeEach(function () {
+                    params = datasets.meanBy1();
+                });
+
+                it('should verify params', function () {
+                    expect(params).to.have.length(1);
+                    expect(params[0] instanceof Array).to.eql(true);
+                });
+
+                it('should find mean value in array', function () {
+                    expect(_.meanBy.apply(_, params)).to.eql(3.25);
+                });
+            });
+
+            /*
+            describe('meanBy - Function', function () {
+                it('should find mean value in function', function () {
+
+                });
+            });
+            //I have no idea how to do it. Probably meanBy doesn't support function. Help!
+            */
+
+            describe('meanBy - Object', function () {
+
+                var params;
+
+                beforeEach(function () {
+                    params = datasets.meanBy3();
+                });
+
+                it('should verify params', function () {
+                    expect(params).to.have.length(2);
+                    expect(params[0] instanceof Object).to.eql(true);
+                    expect(typeof params[1]).to.eql('string');
+                });
+
+                it('should find mean value in object', function () {
+                    expect(_.meanBy.apply(_, params)).to.eql(5);
+                });
+            });
+
+            /*
+            describe('meanBy - String', function () {
+                it('should find mean value in string', function () {
+
+                });
+            });
+
+            //I have no idea how to do it. Probably meanBy doesn't support string. Help!
+            */
+        });
+
+        describe('min', function () {
+
+            var params;
+
+            beforeEach(function () {
+                params = datasets.min();
+            });
+
+            it('should find min value (-3) in user array', function () {
+                expect(params).to.have.length(1);
+                _(params[0]).forEach(function (value) {
+                    expect(typeof value).to.eql('number');
+                });
+                expect(_.min.apply(_, params)).to.eql(-3);
```

Comment:
I can still provide a single number and not to worry about how is this function working.